src/status_im/common/json_rpc/events.cljs

```diff
@@ -83,8 +83,8 @@
                               (.-result response-js)
                               (transforms/js->clj (.-result response-js)))]
                  (if (vector? on-success)
-                   (rf/dispatch (conj on-success result))
-                   (on-success result)))))))))))
+                   (rf/dispatch (conj on-success result (.-id response-js)))
+                   (on-success result (.-id response-js))))))))))))
```

Comment:
@Parveshdhull, the docstring is now outdated.

This part:

```
When a function, it will be called with the transformed response as the sole
argument. When a vector, it is expected to be a valid re-frame event vector,
and the event will be dispatched with the transformed response conj'ed at the
end.
```